File: REMOVE-DUPLICATES/rd.go
package main 

import{
	"fmt"
}

func removeDuplicate(nums [] int) int {
	// i := 1
	// for (int j =0; j<len(nums); j++){
	// 	if(nums[i] != nums[j]){
	// 		nums[i] = nums[j]
	// 		i++
	// 	}
	// }
	set := make(map[int]struct{})
	var tempArr []int

	for(int j = 0; j<len(nums); j++){
		if _, ok := set[nums[j]]; !ok{
			set[nums[j]] = struct{}{}
			tempArr = append(tempArr, nums[j])
		}
	}
	nums = tempArr
	return len(tempArr)
}

func main(){
	nums := [] int {1,1,2}
	fmt.Println(removeDuplicate(nums))
}

File: REMOVE-DUPLICATES/rd.java
import java.util.ArrayList;

public class rd {
    public int removeduplicate(int[] nums){
        int i = 1;
        for (int j = 1; j < nums.length; j++) {
        if(nums[j] != nums[j-1])
        {
            nums[i] = nums[j];
            i++;
        }
        }
        return i;
    }
}